frontend/src/lib/components/Subscriptions/views/EditSubscription.tsx

```diff
@@ -63,14 +72,36 @@ export function EditSubscription({
     }
 
     const commonSelectProps: Partial<LemonSelectProps<LemonSelectOptions>> = {
-        dropdownPlacement: 'top-start',
         type: 'stealth',
         outlined: true,
-        disabled: emailDisabled,
-        dropdownMatchSelectWidth: false,
         dropdownMaxWindowDimensions: true,
     }
 
+    useEffect(() => {
+        if (subscription.target_type === 'slack') {
+            loadSlackChannels()
+        }
+    }, [subscription.target_type])
+
+    // If slackChannels aren't loaded, make sure we display only the channel name and not the actual underlying value
+    const slackChannelOptions = useMemo(
+        () =>
+            slackChannels
+                ? slackChannels.map((x) => ({
+                      key: x.id,
+                      value: `${x.id}|#${x.name}`,
+                      label: x.is_private ? `🔒${x.name}` : `#${x.name}`,
+                  }))
+                : [
+                      {
+                          key: subscription.target_value,
+                          value: subscription.target_value,
+                          label: subscription.target_value?.split('|')?.pop(),
+                      },
+                  ],
+        [slackChannels, subscription.target_value]
+    )
```

Comment:
This could also be a selector

Reply:
Could be. But should it be? It is only used by this component and has some specific logic going on. 

If we used it elsewhere I would agree straight away with pulling it out but otherwise it feels like it would just bloat the logic